src/state_machine/coordinator/frost.rs

```diff
@@ -229,10 +229,18 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
             dkg_id = %self.current_dkg_id,
             "Starting Private Share Distribution"
         );
+        let dkg_public_shares = if self.config.embed_public_private_shares {
+            (0..self.config.num_signers)
+                .map(|id| (id, self.dkg_public_shares[&id].clone()))
```

Comment:
Let's refrain from direct array access.

Reply:
The `frost::Coordinator` requires full participation at every stage of `DKG` and `Sign` operations, so the direct array access here is safe.  

But let's be pedantic anyway :)

Reply:
Yeah, I think the issue is that we need to 100% confident that there is no way for this code to panic. This might be true now, but when we modify some part of this code that may no longer be the case. And if that happens I'd rather return an error than panic.

Reply:
Done.